packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1023,39 +1023,37 @@ class _AppBarState extends State<AppBar> {
     }
     if (leading != null) {
       if (theme.useMaterial3) {
-        if (leading is IconButton) {
```

Comment:
Removing this check might cause a lot of scuba failures like this: https://scuba.corp.google.com/#/report/21d4ef3c-5e05-447e-96ab-6047815fc80c?groupBy=Golden+Location&pageSize=25&pageIndex=0&selected_node=21d4ef3c-5e05-447e-96ab-6047815fc80c&status=2

Checking if the leading is a `IconButton` is originally used to fix the size of leading button size on AppBar([pr link](https://github.com/flutter/flutter/pull/110722)), so we might need to keep this but in order to make the IconTheme work, we can do

```dart
leading = IconButtonTheme(
    data: effectiveIconButtonTheme,
    child: leading is IconButton ? Center(child: leading) : leading,
);
```

Reply:
Ooh great, thank you for the fix! I've added this and verified by running some of the targets in those failing scubas.